sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -173,6 +173,17 @@ private RuntimeException logException(RuntimeException runtimeException, LogLeve
         return runtimeException;
     }
 
+    /*
+     * Determines if the environment and logger support logging at the given log level.
+     *
+     * @param logLevel Logging level for the log message.
+     * @return Flag indicating if logger are configured to support logging at the given log level.
+     */
+    public boolean canLogAtLevel(LogLevel logLevel) {
```

Comment:
Should this be made static and accept a `Class` or `String`? Right now this wouldn't get us too much as almost all use cases of `ClientLogger` are private. From what I understand this is meant to allow someone to determine with their own logic whether or not logging should happen.

Reply:
This is to enable the pattern:

```java
if (logger.canLogAtLevel(ERROR)) {
    logger.error("Output string {}", complexOperationToCalculateValue());
}
```

Reply:
Correct, but I am a bit confused on who this change is targeting, is this for us to use internally where we have our `private final ClientLogger`s or externally from customer code?

Reply:
Correct